server-common/src/main/java/org/apache/gravitino/server/authorization/MetadataFilterHelper.java

```diff
@@ -39,23 +46,84 @@ private MetadataFilterHelper() {}
    * @param metadataList metadata list.
    * @return metadata List that the user has permission to access.
    */
-  @DoNotCall
-  public static NameIdentifier[] filter(
-      MetadataObject.Type metadataType, String privilege, NameIdentifier[] metadataList) {
-    throw new UnsupportedOperationException();
+  public static NameIdentifier[] filterByPrivilege(
+      String metalake,
+      MetadataObject.Type metadataType,
+      String privilege,
+      NameIdentifier[] metadataList) {
+    GravitinoAuthorizer gravitinoAuthorizer =
+        GravitinoAuthorizerProvider.getInstance().getGravitinoAuthorizer();
+    Principal currentPrincipal = PrincipalUtils.getCurrentPrincipal();
+    Entity.EntityType entityType = Entity.EntityType.valueOf(metadataType.name());
+    return Arrays.stream(metadataList)
+        .filter(
+            metaDataName ->
+                gravitinoAuthorizer.authorize(
+                    currentPrincipal,
+                    metalake,
+                    NameIdentifierUtil.toMetadataObject(metaDataName, entityType),
+                    Privilege.Name.valueOf(privilege)))
+        .toArray(NameIdentifier[]::new);
   }
 
   /**
    * Call {@link AuthorizationExpressionEvaluator} to filter the metadata list
    *
+   * @param metalake metalake
    * @param expression authorization expression
    * @param metadataType for example, CATALOG, SCHEMA,TABLE, etc.
    * @param nameIdentifiers metaData list.
    * @return metadata List that the user has permission to access.
    */
-  @DoNotCall
   public static NameIdentifier[] filterByExpression(
-      String expression, MetadataObject.Type metadataType, NameIdentifier[] nameIdentifiers) {
-    throw new UnsupportedOperationException();
+      String metalake,
+      String expression,
+      MetadataObject.Type metadataType,
+      NameIdentifier[] nameIdentifiers) {
+    AuthorizationExpressionEvaluator authorizationExpressionEvaluator =
+        new AuthorizationExpressionEvaluator(expression);
+    return Arrays.stream(nameIdentifiers)
+        .filter(
+            metaDataName -> {
+              Map<MetadataObject.Type, NameIdentifier> nameIdentifierMap =
+                  spiltMetadataNames(metalake, metadataType, metaDataName);
+              return authorizationExpressionEvaluator.evaluate(nameIdentifierMap);
+            })
+        .toArray(NameIdentifier[]::new);
+  }
+
+  /**
+   * Extract the parent metadata from NameIdentify, for example, extract Schema and Catalog from
```

Comment:
Could you optimize the comment?
`NameIdentify` -> `NameIdentifier`.
Could u optimize the words of the comment? Could u explain why we need a map?